src/main/java/com/example/saaca/syncup/model/DueDateScheduler.java

```diff
@@ -110,31 +131,35 @@ public void calculateStartDateAndEndDate(ReturnForm returnForm) {
                 c.setTime(endDate);
                 c.set(currentYear, Calendar.DECEMBER, 31);
                 endDate = c.getTime();
-                c.setTime(currentDate);
-                c.set(currentYear, Calendar.JANUARY, dueDateDay);
+                c.setTime(endDate);
+                c.set(Calendar.DAY_OF_MONTH, 1);
+                c.add(Calendar.MONTH, returnForm.getThirdQuarterMonthOccurrence());
+                if (returnForm.getThirdQuarterDayOccurrence() == 31) {
+                    c.set(Calendar.DAY_OF_MONTH, c.getActualMaximum(Calendar.DAY_OF_MONTH));
+                } else if (c.get(Calendar.MONTH) == Calendar.FEBRUARY && returnForm.getThirdQuarterDayOccurrence() >= 28) {
+                    c.set(Calendar.DAY_OF_MONTH, c.getActualMaximum(Calendar.DAY_OF_MONTH));
+                } else {
+                    c.set(Calendar.DAY_OF_MONTH, returnForm.getThirdQuarterDayOccurrence());
+                }
                 this.setDueDateOfFiling(c.getTime());
             } else {
-                if (returnForm.getReturnType().equals("tds")) {
-                    c.setTime(startDate);
-                    c.set(currentYear, Calendar.JANUARY, 1);
-                    startDate = c.getTime();
-                    c.setTime(endDate);
-                    c.set(currentYear, Calendar.MARCH, 30);
-                    endDate = c.getTime();
-                    c.setTime(currentDate);
-                    c.set(currentYear, Calendar.MAY, dueDateDay);
-                    this.setDueDateOfFiling(c.getTime());
+                c.setTime(startDate);
```

Comment:
we can extract this out in a function ? seems like repeative code

Reply:
haan can do. Will do it in the next PR